File: IdeaProjects/Car Parking Lot System/src/models/DisplayBoard.java
package models;

import Interface.ParkingSpot;

import java.util.HashMap;
import java.util.List;
import java.util.Map;


public class DisplayBoard {
    private int id;
    private Map<String, List<ParkingSpot> > parkingSpots;

    public DisplayBoard(int id){
        this.id = id;
        parkingSpots = new HashMap<>();
    }

    public void showFree(){}
    public void sendParkingFullNotification(){}
    public void addParkingSlot(String spotType, List<ParkingSpot> spots){

    }
}

File: 09_Design Problems/01_Parking Lot System/Essentials/DisplayBoard.cpp
#include <iostream>
#include <map>
#include <list>
#include "../Interfaces/ParkingSpot.cpp"

using namespace std;

class DisplayBoard {
    int id;
    map<string, list<IParkingSpot>> parkingSpots;
    public:
        void updateDisplay() {
            cout << "Display Board Updated" << endl;
        }

        DisplayBoard(int id) {
            this->id = id;
            parkingSpots["Compact"] = list<IParkingSpot>();
        }

        void showFree(){};

        void sendParkingFullNotification() {
            cout << "Parking Lot Full" << endl;
        }

        void addParkingSpot(string type, list<IParkingSpot> spots) {
            cout << "Parking Spot Added to Display Board" << endl;
        }
};